### PR TITLE
Update isOwner props for headers and footers

### LINKED ISSUE #4989: [FxA] Update headers to use abstracted isOwner check
Remove passing accountId  explicitly to headers.  Headers isOwner property should have check for both deviceId or accountId making hasFxA property redundant.

### PR DESCRIPTION
**Pull request by chenba**
Fix #4989, #4990

### CODE REVIEW

File: server/src/pages/shotindex/myshots-header.js [hunk 1]
```diff
@@ -4,13 +4,13 @@ const { SignInButton } = require("../../signin-button.js");
...
 exports.MyShotsHeader = function MyShotsHeader(props) {
-  const signin = props.enableUserSettings && props.hasDeviceId ?
+  const signin = props.enableUserSettings ?
     <SignInButton
       isAuthenticated={props.hasFxa} initialPage="shots"
       staticLink={props.staticLink} /> : null;
 
   return (
-    <Header hasLogo={true} isOwner={props.hasDeviceId} hasFxa={props.hasFxa}>
+    <Header hasLogo={true} isOwner={true} hasFxa={props.hasFxa}>
```
Comment (chenba) [inline server/src/pages/shotindex/myshots-header.js]: My Shots is for the owner, so `isOwner` is simply true here.

File: server/src/pages/shot/model.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ exports.createModel = function(req) {
     id: req.shot.id,
     productName: req.config.productName,
     isExtInstalled: !!req.deviceId,
-    isOwner: req.deviceId === req.shot.ownerId || isFxaAuthenticated,
+    isOwner: req.shot.isOwner,
```
Comment (punamdahiya) [inline server/src/pages/shot/model.js]: I believe we don't have isOwner property available with shot right now. Is this PR supposed to work after work in #4966 
https://github.com/mozilla-services/screenshots/pull/4966/commits/529ff97b7e7308e449a45943ced058fffa09a632#diff-a8b7d067b30e7dfc2ab7f8ba8f8b2b9d

Comment (chenba) [reply server/src/pages/shot/model.js]: Good catch.  I thought that `isOwner` getter change has already landed.  Let's put this PR on hold then.

**Comment by chenba** [comment]
On hold until #4966's merged.

**Comment by punamdahiya** [comment]
Including fix in #4966

**Pull request closed without merging**